src/libmain/progress-bar.cc

```diff
@@ -153,7 +164,7 @@ class ProgressBar : public Logger
         state->activitiesByType[type].its.emplace(act, i);
 
         if (type == actBuild) {
-            auto name = storePathToName(getS(fields, 0));
+            auto name = getName(fields, 4);
```

Comment:
This no longer abbreviates the name, i.e. the progress bar now looks like this:
```
[1/0/2 built, 16.7 MiB DL] building /nix/store/g29xgw3fy960ghskj45x4x1mjv6jv3l2-nix-tarball-2.4pre7196_d94cc07 (autoconfPhase): autoreconf: running: ...
```

Reply:
Indeed, this was happening when using an older daemon (because the messages sent back by the daemon have been extended to include the (abbreviated) derivation name, but there's a fallback code for old daemons that don't do it, but it was missing a call to `storePathToName`).
Should be good now